syntax = "proto3";
package telemetry;
import "google/protobuf/timestamp.proto";
import "uuid.proto";

message GetGestures {
  google.protobuf.Timestamp last_time_sync = 1;
  repeated Gesture gestures = 2;
}

message SaveGesture {
  google.protobuf.Timestamp time_sync = 1;
  Gesture gesture = 2;
}

message DeleteGesture {
  google.protobuf.Timestamp time_sync = 1;
  uuid.UUID id = 2;
}

message PerformGestureById {
  uuid.UUID id = 1;
}

message PerformGestureRaw {
  Gesture gesture = 1;
}

message SetPositions {
  int32 pointer_finger_position = 1;
  int32 middle_finger_position = 2;
  int32 ring_finger_position = 3;
  int32 little_finger_position = 4;
  int32 thumb_finger_position = 5;
}

message Gesture {
  uuid.UUID id = 1;
  string name = 2;
  google.protobuf.Timestamp last_time_sync = 3;
  bool iterable = 4;
  int32 repetitions = 5;
  repeated GestureAction actions = 6;
}

message GestureAction {
  int32 pointer_finger_position = 1;
  int32 middle_finger_position = 2;
  int32 ring_finger_position = 3;
  int32 little_finger_position = 4;
  int32 thumb_finger_position = 5;
  int32 delay = 6;
}